index.js

```diff
@@ -89,17 +158,25 @@ function session(options){
     , cookie = options.cookie || {}
     , trustProxy = options.proxy
     , storeReady = true
-    , rollingSessions = options.rolling || false;
+    , rollingSessions = options.rolling || false
+    , sessionId = options.sessionId || {};
+
+  sessionId.set = sessionId.set || exports.sessionId.set;
+  sessionId.get = sessionId.get || exports.sessionId.get;
+  sessionId.generate = sessionId.generate || options.genid || exports.sessionId.generate;
+
   var resaveSession = options.resave;
   var saveUninitializedSession = options.saveUninitialized;
   var secret = options.secret;
 
-  var generateId = options.genid || generateSessionId;
-
-  if (typeof generateId !== 'function') {
+  if (typeof sessionId.generate !== 'function') {
```

Comment:
This change will cause a misleading error for the case of `sessionId: {generate: 'foo'}`, because the user didn't pass in the `genid` option and will think the correct solution will be to try `sessionId: {generate: 'foo'}, genid: function(){}`, which wouldn't fix the problem.